### PR TITLE
advancedtls: make example to use new pemfile API

### PR DESCRIPTION
**Pull request by ZhenLian**
This PR aims to make the example in advancedtls to use the new pemfile API, which is moved out of adcancedtls and becomes part of gRPC API.
Ideally, the example should base on a main gRPC version, but since the pemfile API is still experimental, it is better to depend on the HEAD to make sure the examples are not out of date(similar to advancedtls). Once all the experimental APIs become stable, we can change to depend on a main gRPC version.
@easwars Could you please take a look? Thank you so much!

### CODE REVIEW

**Comment by dfawley** [comment]
I'm confused.  How is the `examples_test.sh` passing right now if this code doesn't compile as-is?

File: security/advancedtls/examples/go.mod [hunk 1]
```diff
@@ -5,5 +5,10 @@ go 1.15
 require (
 	google.golang.org/grpc v1.33.1
 	google.golang.org/grpc/examples v0.0.0-20201020200225-9519efffeb5d
-	google.golang.org/grpc/security/advancedtls v0.0.0-20201020200225-9519efffeb5d
+	google.golang.org/grpc/security/advancedtls v0.0.0-20201102215344-15ae9fc2b247
```
Comment (dfawley) [inline security/advancedtls/examples/go.mod:8]: This should also get a `replace` directive, so we know the examples are in sync with HEAD.

Comment (ZhenLian) [reply security/advancedtls/examples/go.mod:8]: Done. Could you please take a look again? Thank you so much!

File: security/advancedtls/examples/go.mod [hunk 2]
```diff
@@ -5,5 +5,12 @@ go 1.15
...
 )
+
+// TODO(Zhenlian): remove these lines when advancedtls becomes stable.
```
Comment (dfawley) [inline security/advancedtls/examples/go.mod:11]: I think you should just remove this line altogether.  It's best to know these examples are in sync with HEAD for the whole repo.

File: security/advancedtls/go.mod [hunk 3]
```diff
@@ -8,6 +8,7 @@ require (
 	google.golang.org/grpc/examples v0.0.0-20201020200225-9519efffeb5d
 )
 
+// TODO(Zhenlian): remove these lines when advancedtls becomes stable.
```
Comment (dfawley) [inline security/advancedtls/go.mod:11]: Same.

**Review by dfawley — APPROVED** [review]

**Comment by ZhenLian** [comment]
Thanks for the review! Merging now...

**Pull request merged**